src/reflect/value.go

```diff
@@ -870,31 +1002,118 @@ func (v Value) checkAddressable() {
 	}
 }
 
+func (v Value) checkRO() {
+	if v.isRO() {
+		panic("reflect: value is not settable")
+	}
+}
+
+// OverflowInt reports whether the int64 x cannot be represented by v's type.
+// It panics if v's Kind is not Int, Int8, Int16, Int32, or Int64.
 func (v Value) OverflowInt(x int64) bool {
-	panic("unimplemented: reflect.OverflowInt()")
+	switch v.Kind() {
+	case Int, Int8, Int16, Int32, Int64:
+		bitSize := v.typecode.Size() * 8
+		trunc := (x << (64 - bitSize)) >> (64 - bitSize)
+		return x != trunc
+	}
+	panic(&ValueError{Method: "reflect.Value.OverflowInt", Kind: v.Kind()})
 }
 
+// OverflowUint reports whether the uint64 x cannot be represented by v's type.
+// It panics if v's Kind is not Uint, Uintptr, Uint8, Uint16, Uint32, or Uint64.
 func (v Value) OverflowUint(x uint64) bool {
-	panic("unimplemented: reflect.OverflowUint()")
+	k := v.Kind()
+	switch k {
+	case Uint, Uintptr, Uint8, Uint16, Uint32, Uint64:
+		bitSize := v.typecode.Size() * 8
+		trunc := (x << (64 - bitSize)) >> (64 - bitSize)
+		return x != trunc
+	}
+	panic(&ValueError{Method: "reflect.Value.OverflowUint", Kind: v.Kind()})
 }
 
 func (v Value) Convert(t Type) Value {
 	panic("unimplemented: (reflect.Value).Convert()")
 }
 
+//go:linkname slicePanic runtime.slicePanic
+func slicePanic()
+
 func MakeSlice(typ Type, len, cap int) Value {
-	panic("unimplemented: reflect.MakeSlice()")
+	if typ.Kind() != Slice {
+		panic("reflect.MakeSlice of non-slice type")
+	}
+
+	ulen := uintptr(len)
+	ucap := uintptr(cap)
+	if ulen > ucap {
+		slicePanic()
+	}
+
+	size, ok := mulNoOverflow(typ.Size(), ucap)
+	if !ok {
+		panic("reflect.MakeSlice: slice size out of range")
+	}
+
+	var slice sliceHeader
+	slice.cap = ucap
+	slice.len = ulen
+	slice.data = alloc(size, nil)
+
+	return Value{
+		typecode: typ.(*rawType),
+		value:    unsafe.Pointer(&slice),
+		flags:    valueFlagExported,
+	}
+}
+
+func mulNoOverflow(x, y uintptr) (uintptr, bool) {
+	if x <= 1 || y <= 1 {
+		return x * y, false
+	}
+	m := x * y
+	return m, x == m/y
 }
 
+var zerobuffer [256]byte
```

Comment:
Hmm, this buffer is kinda large. It's fine of course on Linux for example but it would take up a lot of space in the very limited RAM of microcontrollers (if it's needed, globals are optimized away when they're not needed).
A few suggestion:
* make it smaller (perhaps 32 bytes or so)
* make it read-only. This is a bit more complicated, one way to do that is by using strings (`"\x00\x00\x00" etc) and using the backing array that is generally stored in read-only memory instead of the very precious RAM.

Reply:
I can make a 32-byte `\x00` string and use an init() function to extract the data field and put it in this global.